RealifeTech-CoreSDK/APIUtilities/Tests/APIRequesterHelperTests.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  APIRequesterHelperTests.swift
+//  APIUtilitiesTests
+//
+//  Created by Olivier Butler on 08/10/2020.
+//  Copyright © 2020 Realife Tech. All rights reserved.
+//
+
+import XCTest
+@testable import RealifeTech_CoreSDK
+
+final class APIRequesterHelperTests: XCTestCase {
+
+    private let testDeviceId = "device123"
+    private let testClientId = "client456"
+    private let testClientSecret = "secret789"
+    private let testBaseUrl = "baseUrl123"
+    private var tokenManager: APITokenManagable?
+
+    override func setUp() {
+        super.setUp()
+        tokenManager = APIRequesterHelper.setupAPI(
+            deviceId: testDeviceId,
+            clientId: testClientId,
+            clientSecret: testClientSecret,
+            baseUrl: testBaseUrl)
+    }
+
+    override func tearDown() {
+        tokenManager = nil
+        super.tearDown()
+    }
+
+    func test_setupAPI_staticProperties() {
+        XCTAssertNotNil(tokenManager)
+        XCTAssertNotNil(APIRequesterHelper.tokenManager)
+        XCTAssertTrue(APIRequesterHelper.tokenManager is APITokenManager)
+        XCTAssertEqual(testDeviceId, APIRequesterHelper.deviceId)
+        XCTAssertEqual(testBaseUrl, APIRequesterHelper.baseUrl)
+    }
+
+    func test_setupAPI_oAuthParameters() throws {
+        let requestToTest = OAuthRequester.requestInitialAccessToken()
+        XCTAssertTrue(requestToTest.url?.absoluteString.contains(testBaseUrl) ?? false)
```

Comment:
Instead of `XCTAssertTrue(requestToTest.url?.absoluteString.contains(testBaseUrl) ?? false)`, could we write:
`XCTAssertNotEqualToNil(requestToTest.url?.absoluteString.contains(testBaseUrl))`

I haven't tested, but I think it might be a good idea to use `XCTUnwrap` on the url as well. If not, then the test doesn't tell you whether the test failed because the `url` is nil or because `absoluteString` does not contain `testBaseUrl`.

Reply:
Thanks for the feedback. I don't think there's `XCTAssertNotEqualToNil` in XCTest. If you mean `XCTAssertNil`, it doesn't fully verify if the url contains the base URL. Or we can use `XCTAssertEqual(requestToTest.url?.absoluteString.contains(testBaseUrl), true)` to avoid the default value for safe unwrapping. How do you feel about this?